### PR TITLE
[autodoc] multi_control_multi_target_pauli 

### PR DESCRIPTION
**Pull request by mpharrigan**
Add bloq examples for `multi_control_multi_target_pauli`

### CODE REVIEW

File: qualtran/bloqs/multi_control_multi_target_pauli.py [hunk 1]
```diff
@@ -11,27 +11,29 @@
...
 from qualtran.bloqs.and_bloq import And, MultiAnd
 from qualtran.cirq_interop.t_complexity_protocol import t_complexity, TComplexity
 
 
 @frozen
 class MultiTargetCNOT(GateWithRegisters):
-    """Implements single control, multi-target CNOT_{n} gate in 2*log(n) + 1 CNOT depth.
+    r"""Implements single control, multi-target $C[X^{\otimes n}]$ gate.
+
+    Implements $|0><0| I + |1><1| X^{\otimes n}$ using a circuit of depth $2\log(n) + 1$
```
Comment (fdmalone) [inline qualtran/bloqs/multi_control_multi_target_pauli.py:32]: Go all the way and use langle / rangle?

File: qualtran/bloqs/multi_control_multi_target_pauli.py [hunk 2]
```diff
@@ -44,10 +46,9 @@ def decompose_from_registers(
         self,
         *,
         context: cirq.DecompositionContext,
-        **quregs: NDArray[cirq.Qid],  # type:ignore[type-var]
+        control: NDArray[cirq.Qid],
```
Comment (fdmalone) [inline qualtran/bloqs/multi_control_multi_target_pauli.py:49]: [autodoc] hints at just documentation changes :)

File: qualtran/bloqs/multi_control_multi_target_pauli.py [hunk 3]
```diff
@@ -11,27 +11,29 @@
...
 class MultiTargetCNOT(GateWithRegisters):
-    """Implements single control, multi-target CNOT_{n} gate in 2*log(n) + 1 CNOT depth.
+    r"""Implements single control, multi-target $C[X^{\otimes n}]$ gate.
+
+    Implements $|0><0| I + |1><1| X^{\otimes n}$ using a circuit of depth $2\log(n) + 1$
+    containing only CNOT gates.
 
-    Implements CNOT_{n} = |0><0| I + |1><1| X^{n} using a circuit of depth 2*log(n) + 1
-    containing only CNOT gates. See Appendix B.1 of https://arxiv.org/abs/1812.00954 for
-    reference.
```
Comment (fdmalone) [inline qualtran/bloqs/multi_control_multi_target_pauli.py:34]: Feel free to add Args and Registers too / open an issue.

**Review by fdmalone — APPROVED** [review]
LGTM. You can ignore my comments as maybe they're beyond the scope of the PR.

File: qualtran/bloqs/multi_control_multi_target_pauli.py (hunk 1 above)
Comment (mpharrigan) [reply qualtran/bloqs/multi_control_multi_target_pauli.py:32]: trying to strike a balance between readability in the file vs in the notebook. I've been secretly harboring a desire to add a pre-processor to the notebook rendering to turn |...> into `\rangle`s

**Comment by mpharrigan** [comment]
Yeah, I'm going to merge as-is; other things can be done later

**Pull request merged**